pkg/encoding/v1/encoding.go

```diff
@@ -219,11 +222,81 @@ func (m *Mount) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+type Health struct {
+	ReadinessProbe interface{} `yaml:"readinessProbe,omitempty"`
+	Readiness      *api_v1.Probe
+
+	LivenessProbe interface{} `yaml:"livenessProbe,omitempty"`
+	Liveness      *api_v1.Probe
+}
+
+func interfaceToJSON(i interface{}) interface{} {
+	switch x := i.(type) {
+	case map[interface{}]interface{}:
+		m2 := map[string]interface{}{}
+		for k, v := range x {
+			m2[k.(string)] = interfaceToJSON(v)
+		}
+		return m2
+	case []interface{}:
+		for i, v := range x {
+			x[i] = interfaceToJSON(v)
+		}
+	}
+	return i
+}
+
+func interfaceToProbe(i interface{}) (*api_v1.Probe, error) {
```

Comment:
please add a comment, explaining what is a reason for this function.
Are there any limitations what on how interface should look like?
Is is possible to use something more specific instead of an empty interface?